### ISSUE TITLE
Strange distance/speed values for GSM-tracked boat in UHF-tracked fleet

### ISSUE THREAD

**Issue by aksajhfduwafe** [aksajhfduwafe/bugzilla-to-github-issues-test#968]
| Field | Value |
|---|---|
| **Reporter** | @axeluhl |
| **Assignee** | @axeluhl |
| **Product** | Sailing Race Analytics |
| **Component** | Server |
| **Version** | unspecified |
| **Hardware** | All |
| **OS** | All |
| **Severity** | normal |
| **Priority** | P1 |
| **Created** | 2012-09-15T15:20:26Z |
| **Original Bug** | [Bug 968](https://bugzilla.sapsailing.com/bugzilla/show_bug.cgi?id=968) |

---

At the 49er Europeans Championship 2012, Lake Garda, in the Finals Round 4, the boat with sail ID FRA 8 had a non-working UHF tracker. All other boats were tracked using UHF at a 1Hz sampling rate. The FRA 8 boat instead was sampled at ~0.3Hz. Speed and distance values seem off by a factor of three.

**Comment by aksajhfduwafe** [comment]
**@axeluhl** commented:

Come up with a simulation together with TracTrac. In parallel, write a test case that repeats the same GPS fix three times, with time stamps increasing in 1s steps, resulting in a sudden jump after three fixes, and then see what this does to the DistanceCache.

**Comment by aksajhfduwafe** [comment]
**@axeluhl** commented:

Added a test case testJumpyFixes to TrackTest that investigates the hypothesis that we may have received fixes with ascending time stamps but unchanged positions. The test at least succeeds, not producing any obvious bug in the DistanceCache implementation.

**Comment by aksajhfduwafe** [comment]
**@axeluhl** commented:

Did a test with TracTrac today. It turns out that the problem only occurs if the delay that we use in our requests is shorter than the shortest time between two fixes arriving. As long as we kept the client delay above 3s the problem did not occur. As soon as we lowered the delay to 1s, the distance values for the boat tracked with the GSM tracker started to deviate, again roughly by the factor of 3 that we had observed earlier.

I suspect a bug in DistanceCache in case the requested "to" value is before the next fix.
